client/components/common/UserCustomSets.tsx

```diff
@@ -0,0 +1,466 @@
+/** @jsx jsx */
+
+import React from 'react';
+import { jsx } from '@emotion/core';
+import { useQuery } from '@apollo/client';
+import { useTheme } from 'emotion-theming';
+import InfiniteScroll from 'react-infinite-scroller';
+
+import { Theme } from 'common/types';
+import {
+  userProfileSets,
+  userProfileSetsVariables,
+} from 'graphql/queries/__generated__/userProfileSets';
+import userProfileSetsQuery from 'graphql/queries/userProfileSets.graphql';
+import { Input, Select, Tabs } from 'antd';
+import { useTranslation } from 'i18n';
+import { inputFontSize, itemCardStyle, selected } from 'common/mixins';
+import { mq, DEBOUNCE_INTERVAL } from 'common/constants';
+import Link from 'next/link';
+import {
+  CardTitleWithLevel,
+  BrokenImagePlaceholder,
+  CardSkeleton,
+} from 'common/wrappers';
+import { useRouter } from 'next/router';
+import { useDebounceCallback } from '@react-hook/debounce';
+import Card from 'components/common/Card';
+import { getImageUrl, capitalize } from 'common/utils';
+import { customSetTags } from 'graphql/queries/__generated__/customSetTags';
+import customSetTagsQuery from 'graphql/queries/customSetTags.graphql';
+import classesQuery from 'graphql/queries/classes.graphql';
+import { classes } from 'graphql/queries/__generated__/classes';
+import { TabPane } from 'rc-tabs';
+
+const PAGE_SIZE = 20;
+const THRESHOLD = 600;
+
+interface Props {
+  username: string | null;
```

Comment:
Shouldn't username be required? If it's not provided, we shouldn't be rendering this component at all, right?